### DISCUSSION TITLE
Possible to port to TX81z?

### DISCUSSION THREAD

**Discussion by sadguitarius** [ajxs/yamaha_dx97#5] [General]
I just found this project and it's awesome! I don't have a DX9 but I do have a TX81z. I assume their firmware is pretty similar and I found  [this repo](https://github.com/raybellis/TX81Z) which should at least help somewhat. What do you think the chances are of porting some of these changes over to the TX81z? I must say that I have zero assembler experience and have done very little embedded programming, but I'd be happy to give this a shot as a fun learning project. I do have some DSP experience, so I'm not completely in the dark. Do you have any tips about where to start? I'm mostly interested in trying to get pitch EG working and maybe after that trying to add a reverse sawtooth LFO waveform. Thank you!

**Comment by ajxs** [comment]
Hi @sadguitarius, thanks for checking out the project!
This project is possible because the DX7, and DX9 use the same sound chips internally. The TX81Z uses different hardware internally, and doesn't support 6 operators. You could probably add a pitch EG though! The DX7 pitch EG is implemented in software, so it's theoretically possible to add the same to the TX81Z.

If you're interested in learning how I reverse-engineer vintage synth firmware, I wrote a short 'how-to' guide, which you can read here: https://ajxs.me/blog/Introduction_to_Reverse-Engineering_Vintage_Synth_Firmware.html
Learning 8-bit assembly for the 6303 CPU in the DX7/TX81Z isn't too hard, even if you don't have any previous experience with low-level programming. If you're interested, definitely have a try!

**Reply by sadguitarius** [reply]
Thank you so much, that's super helpful! If there was ever any chance of me figuring this out, it got a lot better thanks to your blog post. We'll see how it goes!
